### ISSUE TITLE
Normal Singleplayer Crash as Spawn is loading

### ISSUE THREAD

**Issue by WilliamCook** [Cazadorsniper/PO3#1412]
Pack Version: 3.3.61
Pack Mode (Normal, Titan, Kappa): Normal
Singleplayer, Multiplayer, or LAN?: Singleplayer

Description: I restarted my game and tried to load my world. Now it acts like its gonna load then crashes. I have loaded all of my backups none of them act differently. I did find a similar issue #1260 in previous report but I cant get MCEdit to work it keeps erroring out. I have tried reinstalling java disabling foam fix etc. I posted in the discord and no one replied. I would really appreciate some help I have been trying to fix this for 8 hours. 

Description: Exception in server tick loop

java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1429)
	at java.util.HashMap$ValueIterator.next(HashMap.java:1458)
	at net.minecraft.world.World.func_147448_a(World.java:1949)
	at net.minecraft.world.chunk.Chunk.func_76631_c(Chunk.java:856)
	at net.minecraftforge.common.chunkio.ChunkIOProvider.syncCallback(ChunkIOProvider.java:109)
	at net.minecraftforge.common.chunkio.ChunkIOExecutor.syncChunkLoad(ChunkIOExecutor.java:94)
	at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:118)
	at net.minecraft.world.gen.ChunkProviderServer.func_186028_c(ChunkProviderServer.java:89)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:135)
	at net.minecraft.server.MinecraftServer.func_71222_d(MinecraftServer.java:344)
	at net.minecraft.server.integrated.IntegratedServer.func_71247_a(IntegratedServer.java:142)
	at net.minecraft.server.integrated.IntegratedServer.func_71197_b(IntegratedServer.java:156)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486)
	at java.lang.Thread.run(Thread.java:745)

Steps to reproduce: I don't think you'll be able to reproduce it unless we know the issue.

Crash log: https://pastebin.com/mNFhW7rB

**Comment by Cazadorsniper** [comment]
What happened before you restarted the game?

**Comment by WilliamCook** [comment]
I was using my draconic evolution fusion crafter and I had just set up the crafting core with 9 plinths around it. I noticed the game was stalling a bit so I decided to restart. by stall I mean it froze for about 3 seconds twice within a minute.

**Comment by WilliamCook** [comment]
Hey there so I finally decided to just disable armor plus and try to load my world and I got it to load. When I was loading it said there was 1 missing entry armorplus:ailments. I did try to re-enable the mod but it just did the same as before, crashed.

**Comment by Cazadorsniper** [comment]
Have you tried activating the errored entity/tileentity configs in forge.cfg?

**Comment by WilliamCook** [comment]
Ya I saw a a comment about those settings and they didn't work. I know it was previously stated that it was a workbench issue but I never had anything armorplus related on my base after I was about to recover my world. Before I could recover it was when I made the champion bench and used it until there was nothing left for me to do with it. It was placed in the same game session and 6 backups happened without issue then I restarted my game. Essentially bricking all my backups. It was removed when I loaded my world without armor plus.
Also I did try to load my recovered save on a clean install of the most recent version of the game and the previous version, with armor plus enabled.

**Comment by WilliamCook** [comment]
Another update I just loaded up my world and had my chat spammed.

[CHAT]    full_frame(@1737,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@1749,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@1752,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@1784,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@1787,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@2318,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@2321,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@2333,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@2336,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@2432,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@2435,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@2909,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@2912,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@2924,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@2927,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@2959,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@2962,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@3493,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@3496,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@3508,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@3511,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@3543,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@3546,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@4077,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@4080,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@4092,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@4095,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@4127,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@4130,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@4661,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@4664,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@4676,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@4679,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@4775,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@4778,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@5252,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@5255,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@5267,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@5270,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@5302,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@5305,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@5836,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@5839,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@5851,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@5854,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@5886,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@5889,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@6420,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@6423,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@6435,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@6438,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@6470,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@6473,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@7004,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@7007,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@7019,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@7022,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@7118,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@7121,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@7595,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@7598,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@7610,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@7613,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@7645,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@7648,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@8179,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@8182,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@8194,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@8197,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@8229,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@8232,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@8763,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@8766,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@8778,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@8781,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@8813,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@8816,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@9347,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@9350,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@9362,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@9365,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@9461,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@9464,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@9938,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@9941,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@9953,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@9956,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@9988,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@9991,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@10522,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@10525,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@10537,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@10540,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@10572,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@10575,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@11106,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@11109,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@11121,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@11124,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@11156,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@11159,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@11690,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@11693,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@11705,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@11708,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@11804,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@11807,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@12281,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@12284,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@12296,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@12299,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@12331,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@12334,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@12865,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@12868,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@12880,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@12883,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@12915,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@12918,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@13449,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@13452,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@13464,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@13467,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@13499,{},{Null,Object[#112],Object[#112],Integer,Object[#114]})\n    full_frame(@13502,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@14033,{},{Null,Object[#112],Object[#112],Integer,Object[#116]})\n    full_frame(@14036,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n    full_frame(@14048,{},{Null,Object[#112],Object[#112],Integer,Object[#49]})\n    full_frame(@14051,{},{Null,Object[#112],Object[#112],Integer,Object[#39]})\n

I imagine this is because of the advancements and armor plus being gone. not sure if its helpful

**Comment by Cazadorsniper** [comment]
no that is crafttweaker because armorplus is gone, I wonder why the mod is doing that though.

**Comment by Cazadorsniper** [comment]
Does a new world work with armorplus?

**Comment by WilliamCook** [comment]
Ya I had a test world that worked fine.

**Comment by Cazadorsniper** [comment]
.ConcurrentModificationException usually means 2 or more mods are trying to do the same thing at the same time.

**Comment by WilliamCook** [comment]
the only thing that I can think of is the fact that I was fusion crafting with DE and Extended Crafting. That game session was about 12 hours long as I left my game going to farm EMC. Other things I did in that time were, opening dimlet parcels and upgrading my armor from the armor set before bicubic hexa- I cant remember the name all the way to Equivalent Exchanges Gem armor. It doesn't make sense that I wouldn't be able to reenable Armor Plus since everything I had from that mod is gone. Is it possible that this is an NBT Issue with my player data?? I tried using a region fixer and MCEdit. The region fixer said nothing was wrong. MC Edit was having issues but it might have been my lack of experience with Python. Either way it would be super nice to be able to re-enable the mod. I never got to fight the skeleton king. I would be willing to get my save data to you if it will help?

**Comment by WilliamCook** [comment]
Side note: Seriously thank you for your help if I had money I would donate your mod pack is very well done I have enjoyed every minute of it. :)

**Comment by Cazadorsniper** [comment]
you can try removing the realm of insanity folder in the world save file.

**Comment by Cazadorsniper** [comment]
this issue maybe attached to you or an item you're carrying because I think armorplus has nbt data attached to some items.

**Comment by WilliamCook** [comment]
I cant find anything in my current save I should try re-enabling after backing up. Also I was looking at this version of Armor plus on curseforge and I noticed this:

1.12.2 versions and above require TheDragonLib

1.11.2 versions require both Forgelin and TheDragonLib

Any chance that Forgelin could cause it? I tried to look at the dependencies and I didn't see any mods other than Armor Plus but the page says for 1.12.2 it isn't necessary.

I tried to find the realm of insanity folder and it wasn't there. So I looked at the config.

dimensions {
    # The numeral id of the armorplus:arena dimension
    I:arenaDimensionID=120
    B:enableArenaDimension=false

    # Enable/Disable the realm of insanity
    B:enableRealmOfInsanity=false

    # Too unstable, unfinished to be enabled by default.
    B:enableTheArenaDimension=false

    # The numeral id of the armorplus:realm_of_insanity dimension
    I:realmDimensionID=121

**Comment by Cazadorsniper** [comment]
could try without forgelin

**Comment by WilliamCook** [comment]
Sorry it took me awhile to get back to you again. I re-enabled Armor Plus and I was able to load the game. The only armor plus item that had was a placeholder of the manual in my Akashic tome so I removed all the books out of it and made a new one because putting a new book in would just add to the list not fill in the blank. I doubt that was the issue but it is working again. At this point I am just gonna be thankful that it is resolved and hope that it doesn't happen again. I've never had so much trouble with trying to fix a "corrupted" world.

Thank you again for all you help. Keep up the hard work. :)
